components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/util/IdentityDataStoreUtil.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.governance.util;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.claim.metadata.mgt.ClaimMetadataManagementService;
+import org.wso2.carbon.identity.claim.metadata.mgt.exception.ClaimMetadataException;
+import org.wso2.carbon.identity.claim.metadata.mgt.model.LocalClaim;
+import org.wso2.carbon.identity.claim.metadata.mgt.util.ClaimConstants;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.governance.internal.IdentityMgtServiceDataHolder;
+import org.wso2.carbon.user.core.UserCoreConstants;
+import org.wso2.carbon.user.core.UserStoreManager;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Utility class for identity data store related operations.
+ */
+public class IdentityDataStoreUtil {
+
+    private static final Log log = LogFactory.getLog(IdentityDataStoreUtil.class);
+    private static final String EXCLUDED_USER_STORE_DELIMITER = ",";
+
+    public static final String STORE_IDENTITY_CLAIMS = "StoreIdentityClaims";
+
+    private IdentityDataStoreUtil() {}
+
+    /**
+     * Determine whether the given claim needs to be managed in the identity data store.
+     *
+     * @param claimUri     Claim URI under evaluation.
+     * @param tenantDomain Tenant domain.
+     * @param userStoreDomain User store domain.
+     * @return {@code true} if the claim should be stored in the identity data store, {@code false} otherwise.
+     */
+    public static boolean isManagedInIdentityDataStore(String claimUri,
```

Comment:
This is determined by considering claim information. Hence it would be better to reflect that in method name.
The server level and user store level config was considered before hand.

Reply:
isManagedInUserStoreByClaimConfiguration (Consider further optimized version)

As store in identity store word is not promoted but store in user store is the phrase used, we can stick into that.